File: practicas/P1187A.cpp
// stickers and toys - codeforces
// 

#include <iostream>
using namespace std;

void work(int, int, int, int *);

int main()
{
  int cp;
  cin >> cp;
  for (int l = 0; l < cp; l++)
  {
    int n, s, t;
    cin >> n >> s >> t;
    int z[3] = {};
    work(n, s, t, z);
    // cout << z[0] << ", " << z[1] << ", " << z[2] << endl;
    cout << max(z[0], z[1]) + 1 << endl;
  }
}

void work(int e, int s, int t, int *arr)
{
  cout << "e: " << e << ", s: " << s << ", t: " << t << endl;
  if (e > 0)
  {
    if (t == s)
    {
      if (e == t + s)
      {
        arr[0] += s;
        arr[1] += t;
        work(0, 0, 0, arr);
      }
      else if (e == t)
      {
        arr[2]+=e;
        work(e - e, s - e, t - e, arr);
      }
      else
      {
        arr[0]++;
        arr[1]++;
        work(e - 2, s - 1, t - 1, arr);
      }
    }
    else
    {
      int m = (e + 1) / 2;
      if (t > s)
      {
        // cout << "m1: " << m << endl;
        arr[1] += m;
        work(e - m, s, t - m, arr);
      }
      else
      {
        // cout << "m0: " << m << endl;
        arr[0] += m;
        work(e - m, s - m, t, arr);
      }
    }
  }
}

File: practicas/Java/P1187A.java

// Stickers and Toys - codeforces
// 

import java.math.BigInteger;
import java.util.Scanner;

public class P1187A {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int q = sc.nextInt();
    while (q > 0) {
      int n = sc.nextInt();
      int s = sc.nextInt();
      int t = sc.nextInt();
      if (s == t && s == n) {
        System.out.println(1);
      } else if (s == t && s == n / 2) {
        System.out.println(s + 1);
      } else {
        System.out.println(getEggs(n, s, t));
      }
      q--;
    }
  }

  static BigInteger getEggs(int n, int s, int t) {
    int a = 0;
    while (s + t + a != n) {
      s--;
      t--;
      a++;
    }
    // System.out.println(s + " " + t + " " + a);
    BigInteger m = (s >= t) ? BigInteger.valueOf(s).add(BigInteger.ONE) : BigInteger.valueOf(t).add(BigInteger.ONE);
    return m;
  }
}
